src/DependencyInjection/SonataMediaExtension.php

```diff
@@ -461,8 +461,16 @@ public function configureFilesystemAdapter(ContainerBuilder $container, array $c
             $container->removeDefinition('sonata.media.filesystem.mogilefs');
         }
 
-        if ($container->hasDefinition('sonata.media.adapter.filesystem.opencloud') &&
-            (isset($config['filesystem']['openstack']) || isset($config['filesystem']['rackspace']))) {
+        // NEXT_MAJOR: Remove this section and everything related to openstack/rackspace
+        if (
+            $container->hasDefinition('sonata.media.adapter.filesystem.opencloud') &&
+            (isset($config['filesystem']['openstack']) || isset($config['filesystem']['rackspace']))
+        ) {
+            @trigger_error(
+                'Integration with OpenStack / Rackspace is deprecated without replacement since sonata-project/media-bundle 3.x',
```

Comment:
```suggestion
                'Integration with OpenStack / Rackspace is deprecated without replacement since sonata-project/media-bundle 3.x.',
```